lib/jerboa/client/worker.ex

```diff
@@ -343,4 +386,79 @@ defmodule Jerboa.Client.Worker do
     relay.permissions
     |> Enum.each(fn p -> Process.cancel_timer(p.timer_ref) end)
   end
+
+  ## Subscriptions
+
+  @spec add_subscription(state, Client.ip, pid, reference) :: state
+  defp add_subscription(state, peer_addr, sub_pid, sub_ref) do
+    new_subscriptions =
+      state.subscriptions
+      |> Map.update(peer_addr, %{sub_pid => sub_ref}, fn subs ->
+        update_subscriber_ref(subs, sub_pid, sub_ref)
+      end)
+    %{state | subscriptions: new_subscriptions}
+  end
+
+  @spec update_subscriber_ref(subscribers, pid, reference) :: subscribers
+  defp update_subscriber_ref(subscribers, sub_pid, sub_ref) do
+    if Map.has_key?(subscribers, sub_pid) do
```

Comment:
`Map.has_key?` followed by `Map.get` - we traverse the map twice, while `Map.fetch`/`Map.get_and_update` would do the job by traversing just once. Not a big deal (before we load test), but not complex to fix either.